pkgs/top-level/java-packages.nix

```diff
@@ -227,7 +227,7 @@ in {
       ../development/compilers/openjdk/22.nix
       ../development/compilers/zulu/22.nix
       {
-        openjdk22-bootstrap = temurin-bin.jdk-21;
+        openjdk22-bootstrap = temurin-bin.jdk-22;
```

Comment:
Not a big deal, but I think we should keep the version at 21 here, to align with the recommended version in the [OpenJDK build documentation](https://openjdk.org/groups/build/doc/building.html#boot-jdk-requirements):
> ... the boot JDK for building JDK major version N should be a JDK of major version N-1

Reply:
All recent versions of Java build with major version N, so this keeps it consistent.

Reply:
Do we know why? Seems to make more sense to bootstrap with `N-1` tbh...

Reply:
If the concern is to keep it consistent, then we should use `N-1` everywhere. Using the same version introduces a cyclic dependency. Also, it makes sense to follow the recommendation from OpenJDK build docs.

Reply:
I'm not sure how it introduces a cyclic dependency, because the binary package has no dependency on Java.

IMO LTS releases should definitely be built with the `N` version, because otherwise it introduces a dependency on a package that will be marked insecure. Non-LTS, either way works.

Reply:
I hadn't considered the dependency on an EOL JDK. That's a good point.

 By cyclic dependency, I mean that Temurin depends on the same OpenJDK source code that the derivation does.